Arcade - Add new keywords and function names to be on par with version 1.24

ekenes opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

<!-- Please link to a related issue below. -->
<!-- ie, `Resolves #1234`, etc... so GitHub can magically link it -->

Resolves #3873

### Changes
<!--- Describe your changes -->

Adds all the keywords, functions, and profile names listed in #3873 for proper syntax highlighting.

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`

joshgoebel commented on src/languages/arcade.js:
```diff
@@ -223,14 +246,14 @@ export default function(hljs) {
   const SYMBOL = {
     className: 'symbol',
-    begin: '\\$[datastore|feature|layer|map|measure|sourcefeature|sourcelayer|targetfeature|targetlayer|value|view]+'
+    begin: '\\$[datastore|feature|layer|map|measure|sourcefeature|sourcelayer|targetfeature|targetlayer|value|view|editcontext|originalFeature|featureSet|datapoint|reference|rowindex|record|config|sourcedatastore|targetdatastore|feedfeature|fencefeature|fencenotificationtype|locationupdate|measure|aggregatedFeatures|target|join|analytic]+'
```
I think we want `()` here, not `[]`, do we not?  (yes, I realize it's a mistake that predates your PR).   So we need to fix at least that.

Optimally we'd first declared the list as an array (one item per line) and then use `regex.either` to pull it together in regex form... I'll let you decide if you want to complete that work - you can find examples in other grammars.

joshgoebel commented:
Thanks, this will be awesome once we iron out the one regex issue!

ekenes commented:
Thanks @joshgoebel! I just updated to use `regex.either`. I'm not sure if I used it correctly, but the previous tests I added still pass. Let me know how it looks.

joshgoebel commented on src/languages/arcade.js:
```diff
@@ -251,9 +252,41 @@ export default function(hljs) {
+    "fencenotificationtype",
+    "locationupdate",
+    "measure",
+    "aggregatedFeatures",
```
Any reason not to alpha sort these?

joshgoebel commented:
> I'm not sure if I used it correctly,

Looked good to me.

ekenes commented:
@joshgoebel This was listed under the 11.9 changelog. However, I saw that 11.9 published a couple weeks ago. Which version should I look out for this in?

joshgoebel commented:
Oh sorry I didn't catch this in final review... that changelog entry will need to be moved to 11.10, since 11.9 has passed.
